src/Behat/Testwork/Specification/SpecificationIterator.php

```diff
@@ -18,7 +18,7 @@
  *
  * @author Konstantin Kudryashov <ever.zet@gmail.com>
  *
- * @extends Iterator<int, Suite>
+ * @extends Iterator<int, mixed>
```

Comment:
This annotation was wrong, my mistake, looking at its definition I thought it would implement an iterator of Suites but it actually implements an iterator of specifications which are mixed

Reply:
It should actually be generic instead of using `mixed`, allowing better types for implementations of this interface. For instance, `LazyFeatureIterator` is an iterator over `FeatureNode`, not over `mixed`.
Basically, it should forward the (covariant) template type from `Iterator` for the value type.

`NoSpecificationsIterator` will be a `SpecificationIterator<never>` and `SpecificationArrayIterator` will be generic (inferred from the constructor argument passing the array). `GroupedSpecificationIterator` will also be generic (and its static `group` method as well).

Reply:
You lost me here. What would be your proposed solution?

Reply:
```php
/**
 * @template-covariant T
 * @extends Iterator<int, T>
 */
```

And then updating child classes to define the type of specification they work with.

Reply:
This surpasses my knowledge/understanding of generics, I am in a bit of a loss to what would actually need to go in the child classes, if you can provide some code samples (or a new commit), I would be grateful

Reply:
I started adding generic types for `SpecificationIterator`, and this is quite far-reaching (which is mostly related to the fact that `Behat/Testwork` is written to be agnostic of Gherkin and so does not hardcode that the specification being used is FeatureNode). I'll submit that as a separate PR instead.

Reply:
I submitted it as #1575